src/server_manager/cloud/digitalocean_api.ts

```diff
@@ -124,16 +124,44 @@ class RestApiSession implements DigitalOceanSession {
     // confusing email with their droplet password, which could get mistaken for
     // an invite.
     return this.registerKey_(dropletName, publicKeyForSSH).then((keyId: number) => {
-      return this.request<{droplet: DropletInfo}>('POST', 'droplets', {
-        name: dropletName,
-        region,
-        size: dropletSpec.size,
-        image: dropletSpec.image,
-        ssh_keys: [keyId],
-        user_data: dropletSpec.installCommand,
-        tags: dropletSpec.tags,
-        ipv6: true,
-      });
+      return this.makeCreateDropletRequest(dropletName, region, keyId, dropletSpec);
+    });
+  }
+
+  private makeCreateDropletRequest(
+      dropletName: string, region: string, keyId: number,
+      dropletSpec: DigitalOceanDropletSpecification): Promise<{droplet: DropletInfo}> {
+    let requestCount = 0;
+    const MAX_REQUESTS = 10;
+    return new Promise((fulfill, reject) => {
+      const makeRequestRecursive = () => {
+        ++requestCount;
+        this.request<{droplet: DropletInfo}>('POST', 'droplets', {
+              name: dropletName,
+              region,
+              size: dropletSpec.size,
+              image: dropletSpec.image,
+              ssh_keys: [keyId],
+              user_data: dropletSpec.installCommand,
+              tags: dropletSpec.tags,
+              ipv6: true,
+            })
+            .then(fulfill)
+            .catch((e) => {
+              const ACCOUNT_NOT_YET_FINALIZED_ERROR =
```

Comment:
This seems brittle. Let's check the `id` or even just the status code, i.e. 403.

Reply:
I just updated the error message we are actually getting here to be `XHR forbidden failed with 403: We\'re finalizing your account setup. Please try again in 30 seconds.` (the "XHR forbidden failed with 403: " is actually prepended by the `request` method lower in this file).  Does this seem OK?  At this point in the code we don't have the actual XHR error or object, just the error passed to `Promise.reject` from the `request` method.

Reply:
Ah you don't have that field...okay then how about something a little less brittle, e.g. `message.toLower().contains('finalizing')`?

Then, LGTM.